core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java

```diff
@@ -103,6 +97,64 @@ public static boolean isCharTypeComparable(List<RelDataType> argTypes) {
     return true;
   }
 
+  /**
+   * Checks whether two types or more are byte comparable.
+   *
+   * @return Returns true if all operands are of byte type.
+   */
+  public static boolean isByteTypeComparable(List<RelDataType> argTypes) {
+    assert argTypes != null;
+    assert argTypes.size() >= 2;
+    List<RelDataType> argTypes2 = filterAnyAndNull(argTypes);
+    for (Pair<RelDataType, RelDataType> pair : Pair.adjacents(argTypes2)) {
+      RelDataType t0 = pair.left;
```

Comment:
nit: Can probably remove these intermediate variables and just reference `pair.left` and `pair.right` directly below.